frontend/model/contracts/group.js

```diff
@@ -256,6 +334,46 @@ DefineContract({
         archiveProposal(state, data.proposalHash)
       }
     },
+    'gi.contracts/group/removeMember': {
+      validate: (data) => {
+        return objectOf({
+          member: string,
+          memberID: string,
+          groupID: string,
+          // In case it happens in a big group (by proposal)
+          // we need to validate the associated proposal.
+          proposalHash: optional(string),
+          proposalPayload: optional(objectOf({
+            secret: string // NOTE: simulate the OP_KEY_* stuff for now
+          }))
+        })
+      },
+      process (state, { data, meta }) {
+        const rootState = sbp('state/vuex/state')
+        if (data.member === rootState.loggedIn.username) {
+          // If this member is re-joining the group, ignore the rest
+          // so the member doesn't remove themself again.
+          if (sbp('okTurtles.data/get', 'JOINING_GROUP')) {
+            return false
+          }
+        }
+
+        // QUESTION: Shouldn't these validatons be on validate() method above?
+        // But there we don't have access to state and meta.
+
+        // If we do a validation here and it fails,
+        // will this try to ban/propose to remove only "meta.username" <-- i guess!
+        // or everyone in the group who proccesses this? <-- I saw this happening O.o
+
+        // Well... meanwhile I added validateRemoveMember
+        // on the component side too, before group/removeMember.
+        validateRemoveMember(state, { data, meta })
+        Vue.delete(state.profiles, data.member)
```

Comment:
@taoeffect this was my major question in this whole thing related with validations.

As far as I understand, we should have validation on the Vue part (UI) and also here in case someone tries to send a message hardcoded.

But if `/validate` fails, the message won't be sent, correct? Or at least it shouldn't I guess?...

Well, my question here is: When should the validation be done? Multiple places? If so, that's why I created `validateRemoveMember ` and call it on the multiple places before calling `group/removeMember`.

Not sure if it's the correct way though.